app/controllers/users_controller.rb

```diff
@@ -10,14 +10,20 @@ def login
     session[:last_point] = request.referrer
     session[:last_point] = params[:next_point] if params[:next_point]
 
-    ensure_one_login_method || return
-
-    if launchpad_login_required?
+    ensure_at_least_one_login_method || return
+
+    if both_login_methods_on?
+      if params[:login_method] == 'launchpad'
+        redirect_to sso_url
+      elsif params[:login_method] == 'urs'
+        redirect_to cmr_client.urs_login_path
+      else
+        redirect_to root_url
```

Comment:
this is the line sending users back to the front page to login correct?

Reply:
yes. in this case, both login methods are turned on, but somehow the user got here without choosing if they wanted to log in with urs or launchpad, so we send them back to choose.